qiskit_experiments/visualization/plotters/base_plotter.py

```diff
@@ -0,0 +1,453 @@
+    data, and as curve names for a curve-fit. Figure data is not associated with a series or curve and is
+    instead only associated with the figure. Examples include analysis reports or other text that is
+    drawn onto the figure canvas.
+
+    # TODO: Add example usage and description of options and plot-options.
+    """
+
+    def __init__(self, drawer: BaseDrawer):
+        """Create a new plotter instance.
+
+        Args:
+            drawer: The drawer to use when creating the figure.
+        """
+        # Data to be plotted, such as scatter points, interpolated fits, and confidence intervals
+        self._series_data: Dict[str, Dict[str, Any]] = {}
+        # Data for figure-wide drawing, unrelated to series data, such as text or fit reports.
+        self._figure_data: Dict[str, Any] = {}
+
+        # Options for the plotter
+        self._options = self._default_options()
+        # Plotter options that have changed, for serialization.
+        self._set_options = set()
+
+        # Plot options that are updated in the drawer when `plotter.figure()` is called
+        self._plot_options = self._default_plot_options()
+        # Plot options that have changed, for serialization.
+        self._set_plot_options = set()
+
+        # The drawer backend to use for plotting.
+        self._drawer = drawer
+
+    @property
+    def drawer(self) -> BaseDrawer:
+        """The drawer being used by the plotter."""
+        return self._drawer
+
+    @drawer.setter
+    def drawer(self, new_drawer: BaseDrawer):
+        """Set the drawer to be used by the plotter."""
+        self._drawer = new_drawer
+
+    @property
+    def figure_data(self) -> Dict[str, Any]:
+        """Data for the figure being plotted.
+
+        Figure data includes text, fit reports, or other data that is associated with the figure as a
+        whole and not an individual series.
+        """
+        return self._figure_data
+
+    @property
+    def series_data(self) -> Dict[str, Dict[str, Any]]:
+        """Data for series being plotted.
+
+        Series data includes data such as scatter points, interpolated fit values, and
+        standard-deviations. Series data is grouped by series-name and then by a data-key, both strings.
+        Though series data can be accessed through :meth:`series_data`, it is recommended to use
+        :meth:`data_for` and :meth:`data_exists_for`.
+
+        Returns:
+            dict: A dictionary containing series data.
+        """
+        return self._series_data
+
+    @property
+    def series(self) -> List[str]:
+        """Series names that have been added to this plotter."""
+        return list(self._series_data.keys())
+
+    def data_keys_for(self, series_name: str) -> List[str]:
+        """Returns a list of data-keys for the given series.
+
+        Args:
+            series_name: The series name for the given series.
```

Comment:
This reads in a circular fashion.
```suggestion
            series_name: The series name for which to return the data-keys, i.e., the types of data for each series.
```